File: src/com/suushiemaniac/cubing/bld/model/enumeration/piece/LetterPairImage.java
package com.suushiemaniac.cubing.bld.model.enumeration.piece;

import com.suushiemaniac.cubing.alglib.lang.ImageStringReader;
import com.suushiemaniac.cubing.alglib.lang.NotationReader;

public enum LetterPairImage implements PieceType {
	ANY("<any>"), NOUN("NN"), VERB("VV"), ADJECTIVE("ADJ");

	public static LetterPairImage fromToken(String token) {
		for (LetterPairImage img : values()) {
			if (img.mnemonic().equals(token)) {
				return img;
			}
		}

		return null;
	}

	String token;

	LetterPairImage(String token) {
		this.token = token;
	}

	@Override
	public int getTargetsPerPiece() {
		return 1;
	}

	@Override
	public int getNumPieces() {
		return 26;
	}

	@Override
	public int getNumPiecesNoBuffer() {
		return 25; // TODO 26 here as well??
	}

	@Override
	public int getNumAlgs() {
		return 1;
	}

	@Override
	public String humanName() {
		return "LPI";
	}

	@Override
	public String mnemonic() {
		return this.token;
	}

	@Override
	public NotationReader getReader() {
		return new ImageStringReader();
	}

	@Override
	public String toString() {
		return this.mnemonic();
	}
}

File: src/com/suushiemaniac/cubing/bld/model/enumeration/piece/LetterPairImage.kt
package com.suushiemaniac.cubing.bld.model.enumeration.piece

import com.suushiemaniac.cubing.alglib.lang.ImageStringReader
import com.suushiemaniac.cubing.alglib.lang.NotationReader

enum class LetterPairImage(private val token: String) : PieceType {
    ANY("<any>"), NOUN("NN"), VERB("VV"), ADJECTIVE("ADJ");

    override val targetsPerPiece: Int
        get() = 1

    override val numPieces: Int
        get() = 26

    override val numAlgs: Int
        get() = 1

    override val reader: NotationReader
        get() = ImageStringReader()

    override val humanName: String
        get() = "LPI"

    override val mnemonic: String
        get() = this.token

    override fun toString(): String {
        return this.mnemonic
    }

    companion object {
        fun fromToken(token: String): LetterPairImage? {
            for (img in values()) {
                if (img.mnemonic == token) {
                    return img
                }
            }

            return null
        }
    }
}
